File: app/src/main/java/ru/konstantin/notebook/repository/NoteRepository.java
package ru.konstantin.notebook.repository;

import java.util.Date;
import java.util.List;

import ru.konstantin.notebook.entity.Note;

public interface NoteRepository {
    void getNotes(Callback<List<Note>> callback);
    void clear();
    void add(Note note, Callback<Note> callback);

    void remove(Note note, Callback<Note> callback);

    Note edit(Note note, Callback<Note> callback);

}

File: app/src/main/java/ru/konstantin/notebook/repository/NoteRepository.kt
package ru.konstantin.notebook.repository

import ru.konstantin.notebook.entity.Note

interface NoteRepository {
    fun getNotes(callback: Callback<List<Note>>)
    fun clear()
    fun add(note: Note, callback: Callback<Note>)
    fun remove(note: Note, callback: Callback<Note>)
    fun edit(note: Note?, callback: Callback<Note?>?): Note?
}